code/obj/machinery/computer/robotics.dm

```diff
@@ -22,6 +37,121 @@
+						if(message.client)
+							boutput(message, SPAN_ALERT("<b>AI Killswitch process activated.</b>"))
+							boutput(message, SPAN_ALERT("<b>Killswitch will engage in 3 minutes.</b>"))
+						ai_player.killswitch = TRUE
+						ai_player.killswitch_at = TIME + 3 MINUTES
+					else
+						boutput(usr, SPAN_ALERT("Access Denied."))
+			if ("stop_ai_killswitch")
+				var/mob/living/silicon/ai/ai_player = by_type[/mob/living/silicon/ai][params["index"]]
+				ai_player.killswitch_at = 0
+				ai_player.killswitch = FALSE
+				var/mob/message = ai_player.get_message_mob()
+				message_admins(SPAN_ALERT("[key_name(usr)] has stopped the AI self destruct on [key_name(message, 1, 1)]."))
+				logTheThing(LOG_COMBAT, usr, "has stopped the AI killswitch process on [constructTarget(message,"combat")].")
+				if(message.client)
+					boutput(message, SPAN_NOTICE("<b>Killswitch process deactivated.</b>"))
+
+			if ("start_silicon_killswitch")
+				var/obj/item/card/id/I = usr.equipped()
+				if (istype(I))
+					if(src.check_access(I))
+						var/mob/living/silicon/robot/robot = src.tracked_cyborgs[params["index"]]
+						message_admins(SPAN_ALERT("[key_name(usr)] has activated the robot self destruct on [key_name(robot)]."))
+						logTheThing(LOG_COMBAT, usr, "has activated the robot killswitch process on [constructTarget(robot,"combat")]")
+						if(robot.client)
+							boutput(robot, SPAN_ALERT("<b>Killswitch process activated.</b>"))
+							boutput(robot, SPAN_ALERT("<b>Killswitch will engage in 1 minute.</b>"))
+						robot.killswitch = TRUE
+						robot.killswitch_at = TIME + 1 MINUTE
+					else
+						boutput(usr, SPAN_ALERT("Access Denied."))
+			if ("stop_silicon_killswitch")
+				var/mob/living/silicon/robot/robot = src.tracked_cyborgs[params["index"]]
+				robot.killswitch_at = 0
+				robot.killswitch = 0
+				message_admins(SPAN_ALERT("[key_name(usr)] has stopped the robot self destruct on [key_name(robot, 1, 1)]."))
+				logTheThing(LOG_COMBAT, usr, "has stopped the robot killswitch process on [constructTarget(robot,"combat")].")
+				if(robot.client)
+					boutput(robot, SPAN_NOTICE("<b>Killswitch process deactivated.</b>"))
+			if ("start_silicon_lock")
+				var/mob/living/silicon/robot/robot = src.tracked_cyborgs[params["index"]]
+				if(robot.client)
+					if (robot.emagged)
+						boutput(robot, SPAN_NOTICE("<b>Weapon Lock signal blocked!</b>"))
+						return
+					boutput(robot, SPAN_ALERT("<b>Weapon Lock activated!</b>"))
+				robot.weapon_lock = TRUE
+				robot.weaponlock_time = 120
+				robot.uneq_active()
+				logTheThing(LOG_COMBAT, usr, "has activated [constructTarget(robot,"combat")]'s weapon lock (120 seconds).")
+				for (var/obj/item/roboupgrade/upgrade in robot.contents)
+					if (upgrade.activated)
+						upgrade.activated = 0
+						boutput(robot, SPAN_ALERT("<b>[upgrade] was shut down by the Weapon Lock!</b>"))
+					if (istype(upgrade, /obj/item/roboupgrade/jetpack))
```

Comment:
Are there any other ones that have special handling? e.g. mag boot like one? Shields?

Could use use the `upgrade_deactivate` procs on the upgrades instead, or do those have extra behavior?

Reply:
No clue, this is just from previous code. Assumed no change needed